lib/processor/hosted/cpu.h

```diff
@@ -9,14 +9,14 @@
  * @brief	Hosted environment CPU specific primitives
  */
 
-#include <metal/sys.h>
+#include <zephyr/kernel.h>
 
 #ifndef __METAL_HOSTED_CPU__H__
 #define __METAL_HOSTED_CPU__H__
 
 static inline void metal_cpu_yield(void)
 {
-	metal_wait_usec(10);
+	k_busy_wait(10);
```

Comment:
This file is processor specific. k_busy_wait is for zephy OS.
 Should this API not included in zephyr specific header?
May be I am missing something, but I don't understand how metal_cpu_yield will be implemented for FreeRTOS, or baremetal fw of different platforms.

Reply:
Thanks! now change looks good.

Reply:
This specific call was requested by Zephyr.
They don't want k_yield on the posix environment as it will use 100% of at least 1 cpu.
We don't need 100% of a 4GHz CPU to emulate the actions of an MCU.

"hosted" is supposed to be used when your running on your desktop/laptop.  In Zephyr it is the "posix" board or the equivelent for Windows.  If FreeRTOS has a posix mode why could they not implement metal_wait_usec???

Reply:
It looks like the new version keeps the metal_wait_usec and just moves it.  So this is fine.

Reply:
> This specific call was requested by Zephyr. They don't want k_yield on the posix environment as it will use 100% of at least 1 cpu. We don't need 100% of a 4GHz CPU to emulate the actions of an MCU.
> 
> "hosted" is supposed to be used when your running on your desktop/laptop. In Zephyr it is the "posix" board or the equivelent for Windows. If FreeRTOS has a posix mode why could they not implement metal_wait_usec???

That explains why the tests go into timeout. What I did not test is using `k_yield `with `k_can_yield `protection.
However, for the v2024.10 release, it is better to maintain the legacy behavior in all cases.